subsys/net/lib/nrf_cloud/Kconfig

```diff
@@ -181,18 +181,66 @@ config NRF_CLOUD_AGPS_REQ_CELL_BASED_LOC
 	depends on NRF_CLOUD_AGPS_SINGLE_CELL_ONLY
 	default y
 	help
-		When disabled, nRF Cloud will obtain and store the approximate
-		cell-based location but will not send it to the device.
-		When enabled, nRF Cloud will also send the location to the
-		device.
-
-module = NRF_CLOUD_AGPS
-module-str = nRF Cloud A-GPS
-source "subsys/logging/Kconfig.template.log_config"
+	  When disabled, nRF Cloud will obtain and store the approximate
+	  cell-based location but will not send it to the device.
+	  When enabled, nRF Cloud will also send the location to the
+	  device.
 
 endif # NRF_CLOUD_AGPS
 endmenu
 
+menu "nRF Cloud P-GPS"
+
+config NRF_CLOUD_PGPS
+	bool "Enable nRF Cloud Predicted GPS"
+	depends on NRF9160_GPS
+	depends on MODEM_INFO
+	depends on MODEM_INFO_ADD_NETWORK
+	select STREAM_FLASH_ERASE
+	select SETTINGS
+
+if NRF_CLOUD_PGPS
+
+config NRF_CLOUD_PGPS_PREDICTION_PERIOD
+	int "Desired time in minutes between predictions."
+	range 60 480
+	default 240
+	help
+	  This value is used when requesting predicted GPS data from the
+	  cloud. More frequent predictions will result in more accurate
+	  ephemeris values for any given time, but will take more storage
+	  space in flash.
+
+config NRF_CLOUD_PGPS_NUM_PREDICTIONS
+	int "Desired number of predictions."
+	range 1 48
+	default 42
+	help
+	  This value, together with the prediction period, determines the
+	  overall timespan that the GPS predictions will cover. The
+	  default values allows for one week of predictions.
+
+config NRF_CLOUD_PGPS_DOWNLOAD_FRAGMENT_SIZE
+	int "Fragment size for PGPS downloads."
+	range 128 2048
```

Comment:
what is the largest value that "always" works?
if 2048, why not set that as the default?
if there is some case where 2048 fails?

Reply:
As you know, the size of the headers returned from AWS seems to change over time (in an increasing way). so hard to tell.

Is 1700 for FOTA always working these days?  I had to reduce my default to 1500 to make it work on a sub account.

Reply:
if 1700 is not working on prod then it is a bug and it needs to be lowered.